samples/sqlservices/src/controllers/mainController.ts

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as Utils from '../utils';
+import * as vscode from 'vscode';
+import { ApiWrapper } from '../apiWrapper';
+import SplitPropertiesPanel from './splitPropertiesPanel';
+
+/**
+ * The main controller class that initializes the extension
+ */
+export default class MainController implements vscode.Disposable {
+
+	constructor(protected context: vscode.ExtensionContext, protected apiWrapper?: ApiWrapper) {
+
+	}
+
+	// PUBLIC METHODS //////////////////////////////////////////////////////
+
+	public dispose(): void {
+		this.deactivate();
+	}
+
+	/**
+	 * Deactivates the extension
+	 */
+	public deactivate(): void {
+		Utils.logDebug('Main controller deactivated');
+	}
+
+	public activate(): Promise<boolean> {
+		this.registerSqlServicesModelView();
+		this.registerSplitPanelModelView();
+
+		sqlops.tasks.registerTask('sqlservices.clickTask', (profile) => {
+			vscode.window.showInformationMessage(`Clicked from profile ${profile.serverName}.${profile.databaseName}`);
+		});
+
+		return Promise.resolve(true);
+	}
+
+	private registerSqlServicesModelView(): void {
+		sqlops.dashboard.registerModelViewProvider('sqlservices', async (view) => {
```

Comment:
probably should be separate from dashboard, though we can make those refactors later.

Reply:
Agreed, will take later